### DISCUSSION TITLE
Is "-" Specially Handled in PowerShell?

### DISCUSSION THREAD

**Discussion by taoyouh** [PowerShell/PowerShell#16556] [Q&A]
Why my arguments starting with "-" is not expanded? I looked through the documents but found nothing. Here's the confusing input and output:

```
$a=1
echo a$a # a1
echo -a$a # -a$a
echo -a:a$a # -a: and a1 (2 lines)
```

Where can I find the syntax rules?

I found this when I'm using CMake in PowerShell. I tried to use `cmake -DCMAKE_MODULE_PATH=$(pwd)` but I failed.

**Comment by kasini3000** [comment]
powershell> 
xxx  cmake -DCMAKE_MODULE_PATH=(pwd).path

**Comment by rkeithhill** [comment]
Which version of PowerShell are you using?  You probably want to use `cmake -DCMAKE_MODULE_PATH="$pwd"`.  The syntax rules (such as they are) are a bit different for native apps versus built-in commands.

**Reply by taoyouh** [reply]
I was using Windows PowerShell on Windows Server 2019 (maybe 5.1). Adding the quotes worked. I did make it work using `cmake "-DCMAKE_MODULE_PATH=$(pwd)"`. I'm just confused about the rules.

**Reply by ninmonkey** [reply]
Even though other apps interpret single quotes, `cmd` itself does not parse single quotes `'` as literals, it only interprets `"` There's a explanation here <https://stackoverflow.com/a/24181667/341744>

**Reply by rkeithhill** [reply]
RE docs, the only thing I'm finding is `help about_Command_Syntax` and `help about_parsing` but those don't quite cover what you're after. 🙁
